crates/resolver/src/lib.rs

```diff
@@ -33,58 +33,24 @@
 //! hickory-resolver = "*"
 //! ```
 //!
-//! ## Using the Synchronous Resolver
```

Comment:
I wonder if we need a "how to use this in a synchronous context" example?

Reply:
(See below.)